infrastructure/Dockerfile

```diff
@@ -3,19 +3,27 @@ FROM python:3.10-slim
 # install extra requirements
 RUN apt-get clean
 RUN apt-get update -y
-RUN apt-get install gcc g++ libgeos-dev -y
+RUN apt-get install gcc g++ libgeos-dev -y git
 
-# copy and install files requirements
-COPY ./requirements.txt /app/requirements.txt
-RUN pip install -U pip
-RUN pip install -r /app/requirements.txt
+# copy files requirements
+COPY ./pyproject.toml /app/pyproject.toml
+COPY ./poetry.lock /app/poetry.lock
+COPY ./README.md /app/README.md
+RUN pip install -U pip poetry
+
+# pull sub module
+COPY ./.git /.git
+RUN git submodule update --init --recursive
 
 # set working directory
 WORKDIR /app
 
-# copy files over
+# copy main files
 COPY ./pv_site_production /app/pv_site_production
 COPY ./tests /app/tests
 
+# install files
+RUN poetry install
```

Comment:
If this is for production, we could do

```docker
RUN poetry install --without dev
```

to avoid including the dev dependencies.

Reply:
thanks, ill leave it for the moment, and can add it in later